templates/vue/src/components/__tests__/NodeModal.spec.js

```diff
@@ -0,0 +1,80 @@
+import Vue from "vue"
+import { fireEvent } from "@testing-library/vue"
+import { render } from "@/utils/test"
+
+import oneNodeTapestry from "@/fixtures/one-node.json"
+import NodeModal from "@/components/NodeModal.vue"
+import GravityFormsApi from "@/services/GravityFormsApi"
+import routes, { names } from "@/config/routes"
+
+// Wrapper to add button to trigger NodeModal, otherwise, does not render with visible
+const WrappedNodeModal = Vue.component("wrapped-node-modal", {
+  components: {
+    NodeModal,
+  },
+  template: `<div>
+    <b-button v-b-modal.node-modal>Modal Button</b-button>
+    <node-modal/>
+  </div>`,
+})
+
+jest.mock("@/services/GravityFormsApi", () => {
+  return {
+    ...jest.requireActual("@/services/GravityFormsApi"),
+    exists: jest.fn(),
+    getAllForms: jest.fn(),
+  }
+})
+
+const node = oneNodeTapestry.nodes[0]
+const route = {
+  name: names.MODAL,
+  path: `/nodes/${node.id}`,
+  params: {
+    nodeId: node.id,
+    type: "add",
+    tab: "content",
+  },
+}
+const mocks = {
+  $router: [routes.modal],
+  $route: route,
+}
+
+describe("node modal: content - video", () => {
+  let screen
+
+  beforeEach(async () => {
+    GravityFormsApi.exists.mockImplementation(() => Promise.resolve(false))
+    screen = render(WrappedNodeModal, oneNodeTapestry, { mocks: mocks })

```

Comment:
There's some PRs right now, #834 and #836, that once merged would likely change the way you would write these tests. Neither the `WrappedNodeModal` component nor the route mocks would be necessary. Instead you would change this line of code to:

```suggestion
    screen = render(NodeModal, { fixture: oneNodeTapestry }, (_, _, router) => {
        router.push({
          name: names.MODAL,
          params: { nodeId: node.id, type: "add", tab: "content" }
        }) 
    })
```

This makes it closer to the actual API provided by `@testing-library/vue` and makes the test read a bit better. This change would have to wait until those two PRs get merged in however. What do you think @wynnset? Should we keep this as is for now and change it at a later time?

Reply:
I don't think this is super urgent. I suggest we change the base branch here to `835-improve-test-utils` and wait to merge it in after those PRs are merged in.

Reply:
Awesome, thanks. @shirleywbi I fixed the merge conflicts, can you make the changes to the tests?

Reply:
@narendrasss Updated with your suggestion. Looks like I might need to do further investigation though because `PermissionsTable` might be missing something. Getting `Cannot read property 'includes' of undefined`.